spec/features/cart/checkout_spec.rb

```diff
@@ -1,39 +1,199 @@
 require 'rails_helper'
 
 RSpec.describe 'Cart show' do
+
+  describe 'When I havent added items to my cart' do
+    it 'There is not a link to checkout' do
+      visit "/cart"
+
+      expect(page).to_not have_link("Checkout")
+    end
+  end
+
   describe 'When I have added items to my cart' do
     before(:each) do
-      @mike = Merchant.create(name: "Mike's Print Shop", address: '123 Paper Rd.', city: 'Denver', state: 'CO', zip: 80203)
-      @meg = Merchant.create(name: "Meg's Bike Shop", address: '123 Bike Rd.', city: 'Denver', state: 'CO', zip: 80203)
+      @mike = Merchant.create!(name: "Mike's Print Shop", address: '123 Paper Rd.', city: 'Denver', state: 'CO', zip: 80203)
+      @meg = Merchant.create!(name: "Meg's Bike Shop", address: '123 Bike Rd.', city: 'Denver', state: 'CO', zip: 80203)
+
+      @tire = @meg.items.create!(name: "Gatorskins", description: "They'll never pop!", price: 100, image: "https://www.rei.com/media/4e1f5b05-27ef-4267-bb9a-14e35935f218?size=784x588", inventory: 12)
+      @paper = @mike.items.create!(name: "Lined Paper", description: "Great for writing on!", price: 20, image: "https://cdn.vertex42.com/WordTemplates/images/printable-lined-paper-wide-ruled.png", inventory: 3)
+      @pencil = @mike.items.create!(name: "Yellow Pencil", description: "You can write on paper with it!", price: 2, image: "https://images-na.ssl-images-amazon.com/images/I/31BlVr01izL._SX425_.jpg", inventory: 100)
+
+      2.times do
+        visit "/items/#{@paper.id}"
+        click_on "Add To Cart"
+      end
 
-      @tire = @meg.items.create(name: "Gatorskins", description: "They'll never pop!", price: 100, image: "https://www.rei.com/media/4e1f5b05-27ef-4267-bb9a-14e35935f218?size=784x588", inventory: 12)
-      @paper = @mike.items.create(name: "Lined Paper", description: "Great for writing on!", price: 20, image: "https://cdn.vertex42.com/WordTemplates/images/printable-lined-paper-wide-ruled.png", inventory: 3)
-      @pencil = @mike.items.create(name: "Yellow Pencil", description: "You can write on paper with it!", price: 2, image: "https://images-na.ssl-images-amazon.com/images/I/31BlVr01izL._SX425_.jpg", inventory: 100)
-      visit "/items/#{@paper.id}"
-      click_on "Add To Cart"
       visit "/items/#{@tire.id}"
       click_on "Add To Cart"
-      visit "/items/#{@pencil.id}"
-      click_on "Add To Cart"
+
+      3.times do
+        visit "/items/#{@pencil.id}"
+        click_on "Add To Cart"
+      end
+
       @items_in_cart = [@paper,@tire,@pencil]
     end
 
-    it 'Theres a link to checkout' do
+    it "requires a visitor to login or register before checking out" do
       visit "/cart"
 
-      expect(page).to have_link("Checkout")
+      expect(page).to_not have_link("Checkout")
+      expect(page).to have_content("You must Register or Log In to finish checking out.")
+      expect(page).to have_link("Register")
+      expect(page).to have_link("Log In")
```

Comment:
Maybe be a little more specific about clicking the link in the `#checkout-authorization`

Reply:
Updated nested expectations under within CSS by @LHJE